Terraform Plan and Apply GitHub Action Integrations (#401)
cloudposse/atmos: website/docs/integrations/github-actions/atmos-terraform-apply.md

---
title: Atmos Terraform Apply
sidebar_position: 50
sidebar_label: Terraform Apply
---

> The Cloud Posse GitHub Action "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.

Reviewer: ```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.
```

> Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform planfile from a given S3 bucket using metadata in a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a GitHub Workflow Summary.

Reviewer: ```suggestion
Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform) from a given S3 bucket using metadata stored inside a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
```

> This action is intended to be used with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)
> 

Reviewer: ```suggestion
This action is intended to be used together with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)

## Features

This GitHub Action incorporates superior GitOps support for Terraform by utilizing the capabilities of Atmos, enabling efficient management of large enterprise-scale environments.

* **Implements Native GitOps** with Atmos and Terraform
* **No hardcoded credentials.** Use GitHub OIDC to assume roles.
* **Compatible with GitHub Cloud & Self-hosted Runners** for maximum flexibility. 
* **Beautiful Job Summaries** don't clutter up pull requests with noisy GitHub comments
* **100% Open Source with Permissive APACHE2 License** means you have no expensive subscriptions or long-term commitments.

```

---
title: Atmos Terraform Apply
sidebar_position: 50
sidebar_label: Terraform Apply
---

> The Cloud Posse GitHub Action for "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.

Reviewer: ```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Apply" simplifies provisioning Terraform entirely within GitHub Action workflows. It makes it very easy to understand exactly what happened directly within the GitHub UI.
```

Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform) from a given S3 bucket using metadata stored inside a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).

This action is intended to be used together with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)

## Features

This GitHub Action incorporates superior GitOps support for Terraform by utilizing the capabilities of Atmos, enabling efficient management of large enterprise-scale environments.

> * **Implements Native GitOps** with Atmos and Terraform

Reviewer: ```suggestion
* **Implements Native GitOps** with Atmos and Terraform tightly integrated with GitHub UI
```
